Session/Home/App Review/AppReviewPromptModel.swift

```diff
@@ -22,30 +22,38 @@ extension AppReviewPromptModel {
     
     /// Determines the initial state of the app review prompt.
     static func loadInitialAppReviewPromptState(using dependencies: Dependencies) -> AppReviewPromptState?  {
-        /// Check if incomplete app review can be shown again to user on next app launch
-        let retryCount = dependencies[defaults: .standard, key: .rateAppRetryAttemptCount]
-        
         var promptState: AppReviewPromptState?
         
-        // A buffer of 24 hours
-        let buffer: TimeInterval = 24 * 60 * 60
+        if shouldShowAppReviewAgain(using: dependencies) {
+            promptState = .rateSession
+            
+        } else if dependencies[defaults: .standard, key: .didShowAppReviewPrompt] && !dependencies[defaults: .standard, key: .didActionAppReviewPrompt] {
+            dependencies[defaults: .standard, key: .didShowAppReviewPrompt] = false
+            
+            promptState = .enjoyingSession
+        }
+        
+        return promptState
+    }
+    
+    static func shouldShowAppReviewAgain(using dependencies: Dependencies) -> Bool {
+        /// Check if incomplete app review can be shown again to user on next app launch
+        let retryCount = dependencies[defaults: .standard, key: .rateAppRetryAttemptCount]
 
+        // A buffer of 1 hour
+        let buffer: TimeInterval = 60 * 60
```

Comment:
There is a comment below this that says `This block will execute if the current time is within 24 hours of the retryDate` which will need to be updated to match this new buffer period